python/taichi/lang/transformer.py

```diff
@@ -278,64 +278,102 @@ def visit_For(self, node):
         if node.orelse:
             raise TaichiSyntaxError(
                 "'else' clause for 'for' not supported in Taichi kernels")
-        decorated = isinstance(node.iter, ast.Call) and isinstance(
-            node.iter.func, ast.Attribute) and isinstance(node.iter.func.value, ast.Name) \
-                    and node.iter.func.value.id == 'ti'
-        is_ndrange_for = False
-        is_static_for = False
-        is_grouped = False
+        def is_decorated(iter):
+            return isinstance(iter, ast.Call) \
+                   and isinstance(iter.func, ast.Attribute) \
+                   and isinstance(iter.func.value, ast.Name) \
+                   and iter.func.value.id == 'ti' and (
+                           iter.func.attr == 'static'
+                           or iter.func.attr == 'grouped'
+                           or iter.func.attr == 'ndrange')
+        decorated = is_decorated(node.iter)
+        double_decorated = decorated and len(node.iter.args) == 1 \
+                           and is_decorated(node.iter.args[0])
+        is_ndrange_for = 0
+        is_static_for = 0
+        is_grouped = 0
 
         if decorated:
             attr = node.iter.func
+            # outer decorator
             if attr.attr == 'static':
-                is_static_for = True
+                is_static_for = 1
             elif attr.attr == 'grouped':
-                is_grouped = True
+                is_grouped = 1
             elif attr.attr == 'ndrange':
-                is_ndrange_for = True
+                is_ndrange_for = 1
             else:
                 raise Exception('Not supported')
+            if double_decorated:
+                attr = node.iter.args[0].func
+                # inner decorator
+                if attr.attr == 'static':
+                    is_static_for = 2
```

Comment:
One thing to improve here: if the user does `for i in ti.static(ti.static(range(x))`, we should throw a `TaichiSyntaxError`.